Use strongly typed primitive values in CSS/Style gradient code

weinig opened the pull request:
#### 63aa285e5072b20545b6b994024e09d68be3f1ad
<pre>
Use strongly typed primitive values in CSS/Style gradient code
<a href="https://bugs.webkit.org/show_bug.cgi?id=279721">https://bugs.webkit.org/show_bug.cgi?id=279721</a>

Reviewed by Darin Adler.

This change introduces a new set of primitive values that will be
used to implement CSS property values, eventually hoping to replace
the CSSValue hierarchy completely.

The goals of the new representation are to be:
    - strongly typed
    - value based
    - memory efficient
    - worker safe

At a high level, CSS properties will be modeled as two algebraic
data types (e.g. conceptually compose of std::variant, std::tuple
and scalars):
     - the css value (in a new WebCore::CSS namespace)
     - the style value (in the WebCore::Style namespace)

Replacing CSSPrimitiveValue is a new set of strongly typed numeric
primitives. These have names that mirror the specification like
LengthPercentage or Angle, and have representations in both the
CSS and Style worlds. In the CSS world, these primitive numeric
values are the union of both raw numeric representation and the
simplified calc() representation. In the Style world, these
primitive numeric values are usually just the fully resolved raw
value, but in some case, such as the *Foo*Percentage types, they
can have more complex representations including unresolved
calculation values.

To create aggregate (product) types, as will be necessary for most
properties, tuple like containers are available. The first container,
introduced in this change, is called `SpaceSeparatedTuple` and
represents as the name implies, a space separated tuple of values.
By providing the semantics of serialization in the type, we can
simplify how serialization works.

To create variants (sum) types, std::variant can be used directly,
or wrapped in a struct, if custom operations are needed for it.

For simple cases, say, a property with a space separated set of
LengthPercentages:

    &lt;foo&gt; = [ &lt;length-percentage&gt; &lt;length-percentage&gt; ]

the CSS representation would be:

    CSS::SpaceSeparatedTuple&lt;CSS::LengthPercentage, CSS::LengthPercentage&gt;

and the Style representation would be:

    Style::SpaceSeparatedTuple&lt;Style::LengthPercentage, Style::LengthPercentage&gt;

and no custom logic would need to be written.

However, more complex cases, like &lt;position&gt;, which is included in
this change, where the spec mandates specific rules when resolving
the style, a custom type is needed. For &lt;position&gt;, both a CSS and
Style type are needed. For CSS, its roughly (details of the complex
structure elided):

    namespace WebCore::CSS {
      using TwoComponentPosition = ...
      using FourComponentPosition = ...

      struct Position {
        std::variant&lt;TwoComponentPosition, FourComponentPosition&gt; value;
      };
    }

For Style its roughly:

    namespace WebCore::Style {
      struct Position {
        SpaceSeparatedTuple&lt;LengthPercentage, LengthPercentage&gt; value;
      };
    }

In addition to defining the types, the following functions are needed:

For CSS: `serializationForCSS`, `collectComputedStyleDependencies`, `visitCSSValueChildren`
For Style: `toStyle`, `toCSS`

The two crucial ones are the conversion operators in Style, which implement
the custom logic. For the most part, the ones in CSS can usually forward
to builtin implementations, passing the inner value.

This is just the first step in this process. The initial goal is to rework
the implementations of existing CSSValue subclasses, like gradient subclasses
in this change, and evolve the primitives as we go.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterFunctionDescriptor.h:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSUnits.cpp:
* Source/WebCore/css/CSSUnits.h:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ComputedStyleDependencies.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Evaluation.cpp:
* Source/WebCore/css/calc/CSSCalcTree+NumericIdentity.h:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.h:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Traversal.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcType.cpp:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSAbsoluteColorResolver.h:
* Source/WebCore/css/color/CSSAbsoluteColorSerialization.h:
* Source/WebCore/css/color/CSSColorConversion+Normalize.h:
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/color/CSSColorMixResolver.cpp:
* Source/WebCore/css/color/CSSColorMixResolver.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
* Source/WebCore/css/color/CSSColorMixSerialization.h:
* Source/WebCore/css/color/CSSRelativeColorResolver.h:
* Source/WebCore/css/color/CSSRelativeColorSerialization.h:
* Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/CSSUnresolvedRelativeColor.h:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/color/StyleRelativeColor.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AngleDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentageDefinitions.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+IntegerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NoneDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NumberDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+PercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+RawTypes.h: Removed.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ResolutionDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Symbol.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+SymbolDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+TimeDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.cpp: Removed.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h: Removed.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserOptions.h: Copied from Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h.
* Source/WebCore/css/parser/CSSVariableParser.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
* Source/WebCore/css/values/CSSPosition.cpp: Added.
* Source/WebCore/css/values/CSSPosition.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+EvaluateCalc.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+Serialization.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+Serialization.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.h: Added.
* Source/WebCore/css/values/CSSUnevaluatedCalc.cpp: Added.
* Source/WebCore/css/values/CSSUnevaluatedCalc.h: Added.
* Source/WebCore/css/values/CSSValueTypes.h: Added.
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/platform/FloatConversion.h:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationCategory.cpp:
* Source/WebCore/platform/calc/CalculationCategory.h:
* Source/WebCore/platform/calc/CalculationValue.cpp:
* Source/WebCore/platform/calc/CalculationValue.h:
* Source/WebCore/rendering/style/StyleGradientImage.cpp:
* Source/WebCore/rendering/style/StyleGradientImage.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/AnchorPositionEvaluator.h:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/style/StyleResolveForFont.cpp:
* Source/WebCore/style/TransformOperationsBuilder.cpp:
* Source/WebCore/style/values/StylePosition.cpp: Added.
* Source/WebCore/style/values/StylePosition.h: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.cpp: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.h: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.cpp: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.h: Added.
* Source/WebCore/style/values/StyleValueTypes.h: Added.

Canonical link: <a href="https://commits.webkit.org/284305@main">https://commits.webkit.org/284305@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/888adae8823d66934b55f93fe26891e5a8c07b0f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68998 "83 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48398 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21670 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73078 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20149 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71115 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20004 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/54948 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13397 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72064 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44189 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59566 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35422 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40854 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/16995 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18530 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62803 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74787 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12980 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16590 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62588 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13018 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59649 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62485 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15313 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10468 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4074 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44202 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45275 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46471 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45017 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
This looks fantastic. I didn’t review everything line by line, but I think this is going to be great.

weinig committed:
Use strongly typed primitive values in CSS/Style gradient code
https://bugs.webkit.org/show_bug.cgi?id=279721

Reviewed by Darin Adler.

This change introduces a new set of primitive values that will be
used to implement CSS property values, eventually hoping to replace
the CSSValue hierarchy completely.

The goals of the new representation are to be:
    - strongly typed
    - value based
    - memory efficient
    - worker safe

At a high level, CSS properties will be modeled as two algebraic
data types (e.g. conceptually compose of std::variant, std::tuple
and scalars):
     - the css value (in a new WebCore::CSS namespace)
     - the style value (in the WebCore::Style namespace)

Replacing CSSPrimitiveValue is a new set of strongly typed numeric
primitives. These have names that mirror the specification like
LengthPercentage or Angle, and have representations in both the
CSS and Style worlds. In the CSS world, these primitive numeric
values are the union of both raw numeric representation and the
simplified calc() representation. In the Style world, these
primitive numeric values are usually just the fully resolved raw
value, but in some case, such as the *Foo*Percentage types, they
can have more complex representations including unresolved
calculation values.

To create aggregate (product) types, as will be necessary for most
properties, tuple like containers are available. The first container,
introduced in this change, is called `SpaceSeparatedTuple` and
represents as the name implies, a space separated tuple of values.
By providing the semantics of serialization in the type, we can
simplify how serialization works.

To create variants (sum) types, std::variant can be used directly,
or wrapped in a struct, if custom operations are needed for it.

For simple cases, say, a property with a space separated set of
LengthPercentages:

    <foo> = [ <length-percentage> <length-percentage> ]

the CSS representation would be:

    CSS::SpaceSeparatedTuple<CSS::LengthPercentage, CSS::LengthPercentage>

and the Style representation would be:

    Style::SpaceSeparatedTuple<Style::LengthPercentage, Style::LengthPercentage>

and no custom logic would need to be written.

However, more complex cases, like <position>, which is included in
this change, where the spec mandates specific rules when resolving
the style, a custom type is needed. For <position>, both a CSS and
Style type are needed. For CSS, its roughly (details of the complex
structure elided):

    namespace WebCore::CSS {
      using TwoComponentPosition = ...
      using FourComponentPosition = ...

      struct Position {
        std::variant<TwoComponentPosition, FourComponentPosition> value;
      };
    }

For Style its roughly:

    namespace WebCore::Style {
      struct Position {
        SpaceSeparatedTuple<LengthPercentage, LengthPercentage> value;
      };
    }

In addition to defining the types, the following functions are needed:

For CSS: `serializationForCSS`, `collectComputedStyleDependencies`, `visitCSSValueChildren`
For Style: `toStyle`, `toCSS`

The two crucial ones are the conversion operators in Style, which implement
the custom logic. For the most part, the ones in CSS can usually forward
to builtin implementations, passing the inner value.

This is just the first step in this process. The initial goal is to rework
the implementations of existing CSSValue subclasses, like gradient subclasses
in this change, and evolve the primitives as we go.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFilterFunctionDescriptor.h:
* Source/WebCore/css/CSSFontFaceSet.cpp:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/CSSGradientValue.h:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSUnits.cpp:
* Source/WebCore/css/CSSUnits.h:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ComputedStyleDependencies.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Evaluation.cpp:
* Source/WebCore/css/calc/CSSCalcTree+NumericIdentity.h:
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Parser.h:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Traversal.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcType.cpp:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/color/CSSAbsoluteColorResolver.h:
* Source/WebCore/css/color/CSSAbsoluteColorSerialization.h:
* Source/WebCore/css/color/CSSColorConversion+Normalize.h:
* Source/WebCore/css/color/CSSColorConversion+ToTypedColor.h:
* Source/WebCore/css/color/CSSColorDescriptors.h:
* Source/WebCore/css/color/CSSColorMixResolver.cpp:
* Source/WebCore/css/color/CSSColorMixResolver.h:
* Source/WebCore/css/color/CSSColorMixSerialization.cpp:
* Source/WebCore/css/color/CSSColorMixSerialization.h:
* Source/WebCore/css/color/CSSRelativeColorResolver.h:
* Source/WebCore/css/color/CSSRelativeColorSerialization.h:
* Source/WebCore/css/color/CSSUnresolvedAbsoluteColor.h:
* Source/WebCore/css/color/CSSUnresolvedColor.h:
* Source/WebCore/css/color/CSSUnresolvedColorMix.cpp:
* Source/WebCore/css/color/CSSUnresolvedColorMix.h:
* Source/WebCore/css/color/CSSUnresolvedRelativeColor.h:
* Source/WebCore/css/color/StyleColorMix.h:
* Source/WebCore/css/color/StyleRelativeColor.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Angle.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AngleDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.cpp: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentage.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AnglePercentageDefinitions.h: Added.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+CSSPrimitiveValueResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Integer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+IntegerDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Length.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+LengthPercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaConsumer.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+MetaResolver.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+None.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NoneDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Number.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+NumberDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Percentage.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+PercentageDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Position.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+RawTypes.h: Removed.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Resolution.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ResolutionDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Symbol.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+SymbolDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Time.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+TimeDefinitions.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.cpp: Removed.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+UnevaluatedCalc.h: Removed.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h:
* Source/WebCore/css/parser/CSSPropertyParserOptions.h: Copied from Source/WebCore/css/parser/CSSPropertyParserConsumer+Primitives.h.
* Source/WebCore/css/parser/CSSVariableParser.cpp:
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/css/typedom/CSSNumericValue.cpp:
* Source/WebCore/css/values/CSSPosition.cpp: Added.
* Source/WebCore/css/values/CSSPosition.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+ComputedStyleDependencies.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+EvaluateCalc.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+Serialization.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+Serialization.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.cpp: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes+SymbolReplacement.h: Added.
* Source/WebCore/css/values/CSSPrimitiveNumericTypes.h: Added.
* Source/WebCore/css/values/CSSUnevaluatedCalc.cpp: Added.
* Source/WebCore/css/values/CSSUnevaluatedCalc.h: Added.
* Source/WebCore/css/values/CSSValueTypes.h: Added.
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/platform/FloatConversion.h:
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/calc/CalculationCategory.cpp:
* Source/WebCore/platform/calc/CalculationCategory.h:
* Source/WebCore/platform/calc/CalculationValue.cpp:
* Source/WebCore/platform/calc/CalculationValue.h:
* Source/WebCore/rendering/style/StyleGradientImage.cpp:
* Source/WebCore/rendering/style/StyleGradientImage.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/AnchorPositionEvaluator.h:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/style/StyleResolveForFont.cpp:
* Source/WebCore/style/TransformOperationsBuilder.cpp:
* Source/WebCore/style/values/StylePosition.cpp: Added.
* Source/WebCore/style/values/StylePosition.h: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.cpp: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes+Conversions.h: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.cpp: Added.
* Source/WebCore/style/values/StylePrimitiveNumericTypes.h: Added.
* Source/WebCore/style/values/StyleValueTypes.h: Added.

Canonical link: https://commits.webkit.org/284305@main
